Fix destring

konradzdeb opened the pull request:
Hi, I've fixed a bug in the call to destring as it wasn't evaluating within the data frame environment. I have also fixed a small mistake in `.Rbuildignore`. 

As a wider point, this brings attention to mock tests. Proper tests should catch that but we would have to mock metrics object properly.

konradzdeb committed:
Documention

Introduced internal keyword to hide utility functions from documentation index exposed to user.

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb committed:
Consistent column names

Started working on function to generate consistent column names for metrics objects

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb committed:
Travis warnings

Changed warnings are errors to true as it's recommended setting for submission to CRAN

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb committed:
Merge branch 'travis-appveyor-configs' into fix-consistent-column-names

konradzdeb committed:
Cleaning test file

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb committed:
Merge branch 'master' of https://github.com/glenn-m/promR into fix-consistent-column-names

Updated development branch with recent changes to the master branch.

konradzdeb committed:
Travis

Updated travis configuration, possibly checking for warnings makes sense in the context or R release version and package being available on CRAN.

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb committed:
Metrics raw

Added structure for response like object representing raw metrics to be created

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb committed:
Created helper objects

Following this discussion (https://stackoverflow.com/questions/32328802/where-to-put-data-for-automated-tests-with-testthat) I've created helper object to be used with testthat. It seems that storing small files in test directory is permissible but there is no clarity on naming convention.

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb committed:
Minor: changed -> to <-

As per SO discussion related to another project

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb committed:
Minor: changed -> to <-

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb committed:
Metrics object formatting

This utility function cleans column names and may potentially close #18

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb committed:
Better comment wording

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb committed:
Vignettes

Created skeleten for vignette covering basic usage

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>
#22

konradzdeb committed:
Merge remote-tracking branch 'upstream/master' into vignettes

konradzdeb committed:
Cleaned DESCRIPTION

Added package version for checkmate and ordered alphabetically both
lists.

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb committed:
Merge remote-tracking branch 'upstream/master' into vignettes

konradzdeb committed:
Documentation

Minor progress with vignette development.

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb committed:
Documentation build

Ensured that used version of knitr corresponds to minimum requirements
that are used in production of vignettes #22.

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb committed:
Documentation build

Ignoring vignette cache folder #22.

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb committed:
Vignette development

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb committed:
Update following PRs

Bringing recent master changes to current dev work.

konradzdeb committed:
Fix: ignoring _cahce files

I've made mistake in providing exclusion pattern for _cache/ files
constructed by vignette builder.

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb committed:
Fix: destring call

Fix to the destring call so it's executed within data frame environment.

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb commented on R/format_metrics_data.R:
```diff
@@ -55,16 +55,20 @@ format_metrics_range_data <- function(x) {
+                  }
+                ))
+
+  res <- within(data = res,
```
This is now amended and evaluates within the data object `res`

glenn-m commented:
Can we add in some extra test cases? 

You can see where you could add in two extra tests [here](https://codecov.io/gh/glenn-m/promR/src/a111df49151b92d9f55f190f4018b3f2af08d87e/R/format_metrics_data.R). We should have a test which passes in a factor at least.

Thanks

konradzdeb commented:
> Can we add in some extra test cases?

Yes, definitely worth doing. Also the whole testing mechanism could be upgraded to mock API behaviour, just to test the whole pipeline.

glenn-m commented:
> > Can we add in some extra test cases?
> 
> Yes, definitely worth doing. Also the whole testing mechanism could be upgraded to mock API behaviour, just to test the whole pipeline.

Yeah, I've been trying to think of the best approach to do it. I'll see if I can work out a decent way today, as I'm planning on finishing the metadata query functionality so will need to write tests for it.

konradzdeb committed:
Create separate test files for easier editing and adding more tests (#39)

konradzdeb committed:
Metrics formatting: extra test

Added another test checking if right error is generated for non-data
frame object. This should increase coverage (a little).

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb committed:
Testing: increasing coverage + better names handling

If unnamed object for some reason lands within the rename metrics
function then it returns self instead of NULL - potentially safer if
names would get lost somewhere. Added tests for that behaviour which
should also increase coverage.

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb committed:
Added pkgdown website to .Rbuildignore

Signed-off-by: Konrad Zdeb <konrad.zdeb@me.com>

konradzdeb committed:
Not ignoring docs as this is not the problem
#29

konradzdeb commented:
_Reminder_ problem is in `curl::has_internet` call not being listed as a dependency.

konradzdeb committed:
Updating

Updating this dev branch with upstream master update.

konradzdeb commented:
I reckon testing merits longer re-work so probably better to complete as one PR (#32).
